charm-slurmctld/src/charm.py

```diff
@@ -140,13 +143,26 @@ def _cluster_info(self):
     @property
     def _addons_info(self):
         """Assemble addons for slurm.conf."""
-        # NOTE add prolog and epilog
         # NOTE add elasticsearch
 
-        return {**self._assemble_acct_gather_addons()}
+        return {**self._assemble_prolog_epilog(),
+                **self._assemble_acct_gather_addons()}
+
+    def _assemble_prolog_epilog(self) -> dict:
+        """Generate the prolog_epilog section of the addons."""
+        logger.debug("## Generating prolog epilog configuration")
+
+        prolog_epilog = self._prolog_epilog.get_prolog_epilog()
+
+        if prolog_epilog:
+            return {"prolog_epilog": prolog_epilog}
```

Comment:
How do you feel about in-line `if` expressions:
```python
return {"prolog_epilog": prolog_epilog} if prolog_epilog else {}
```

Reply:
I prefer `if-else` blocks, I think they are more readable.